src/UserService.Business/Commands/Skill/CreateSkillCommand.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Net;
+using System.Collections.Generic;
+using LT.DigitalOffice.Kernel.FluentValidationExtensions;
+
+namespace LT.DigitalOffice.UserService.Business.Commands.Skill
+{
+    public class CreateSkillCommand : ICreateSkillCommand
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IUserRepository _userRepository;
+        private readonly IDbSkillMapper _mapper;
+        private readonly ISkillRepository _skillRepository;
+        private readonly IAccessValidator _accessValidator;
+        private readonly ICreateSkillRequestValidator _validator;
+
+        public CreateSkillCommand(
+            IHttpContextAccessor httpContextAccessor,
+            IUserRepository userRepository,
+            IDbSkillMapper mapper,
+            ISkillRepository skillRepository,
+            IAccessValidator accessValidator,
+            ICreateSkillRequestValidator validator)
+        {
+            _httpContextAccessor = httpContextAccessor;
+            _userRepository = userRepository;
+            _mapper = mapper;
+            _skillRepository = skillRepository;
+            _accessValidator = accessValidator;
+            _validator = validator;
+        }
+
+        public OperationResultResponse<Guid> Execute(CreateSkillRequest request)
+        {
+            OperationResultResponse<Guid> response = new();
+            Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+            DbUser dbUser = _userRepository.Get(senderId);
+
+            if (!(dbUser.IsAdmin || _accessValidator.HasRights(Rights.AddEditRemoveUsers)))
+            {
+                response.Errors.Add("Not enough rights.");
+                response.Status = OperationResultStatusType.Failed;
+                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
+
+                return response;
+            }
+
+            if (!_validator.ValidateCustom(request, out List<string> errors))
+            {
+              _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+              response.Errors.AddRange(errors);
+              response.Status = OperationResultStatusType.Failed;
+
+              return response;
+            }
+
+            if (_skillRepository.DoesSkillAlreadyExist(request.Name))
+            {
+                response.Errors.Add($"Skill {request.Name} already exist");
+                response.Status = OperationResultStatusType.Failed;
+                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Conflict;
+
+                return response;
+            }
+
+            response.Body = _skillRepository.Add(_mapper.Map(request));
+
+            if(response.Body == default)
+            {
+                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Conflict;
+                response.Status = OperationResultStatusType.Failed;
```

Comment:
you do not return, in this case in any way you get crated and fullSuccess status

Reply:
++